### PR TITLE
[Pydantic II] Change all paths to pydantic types

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@h-mayorquin The follow-up replacing all usage with the Pydantic ones

Still need to make my way through the ecephys ones

### CODE REVIEW

File: src/neuroconv/utils/types.py [hunk 1]
```diff
@@ -1,12 +1,7 @@
-from pathlib import Path
-from typing import Optional, TypeVar, Union
+from typing import Optional, Union
 
 import numpy as np
 
-FilePathType = TypeVar("FilePathType", str, Path)
```
Comment (CodyCBakerPhD) [inline src/neuroconv/utils/types.py:6]: Yes, this is a hard deprecation

I could not think of a good way to do a soft one without utilizing module-level `__getattr__` magic to inject a deprecation warning

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin All tests should be green now

**Review by h-mayorquin — APPROVED** [review]
Unrelated but I would like to eliminate the OptionalArray as well

**Comment by CodyCBakerPhD** [comment]
> Unrelated but I would like to eliminate the OptionalArray as well

One step at a time I say 😉

**Comment by CodyCBakerPhD** [comment]
But yes, happy to do that in follow-up

**Pull request merged**